src/winetricks

```diff
@@ -8312,9 +8383,11 @@ load_dotnet35()
         w_package_unsupported_win64
     fi
 
+    # Platform compatibility
     case "$W_PLATFORM" in
-        windows_cmd) ;;
-        *) w_warn "dotnet35 does not yet fully work or install on wine.  Caveat emptor." ;;
+        windows_cygwin|windows_windows) ;;
+        nix_linux|nix_freebsd|nix_darwin|nix_redox) w_warn "dotnet35 does not yet fully work or install on wine.  Caveat emptor." ;;
+        *) w_die "Unknown platform '$W_PLATFORM' parsed in load_dotnet35()"
```

Comment:
I highly doubt this is needed, if you're on an unknown platform it would've died already.

Reply:
if the function is called directly it woudn't die + sanitization

Reply:
What do you mean called directly?

Reply:
@austin987 

```shell
# Assuming W_PLATFORM=""
load_something() { w_try_regedit32 ;} 
```

Script should fail if W_PLATFORM is not exported to prevent unexpected or if something unset W_PLATFORM

Reply:
By the time load_something runs, winetricks_init() would already have run.

Rather than hypotheticals, do you have a testcase demonstrating it?

Reply:
# Kreytricks rebase
i'm working on https://github.com/Winetricks/winetricks/pull/1359 so that winetricks woudn't make a new wineprefix on init so that https://github.com/Winetricks/winetricks/pull/1348 could be used on https://github.com/Winetricks/winetricks/pull/1347 (and others) without the need to reinvoke winetricks which could conflict with it where W_PLATFORM is not exported.

# Hypothetical
Other then that assuming that it supports WSL1, Native Windows and Cygwin i would assume there being forks that tries to unset this to make it working on their system and so it should encourage contrib

# Practical
<img width="810" alt="image" src="https://user-images.githubusercontent.com/11302521/66441184-0dabb080-ea36-11e9-9fa7-5161259cd7a0.png">
<img width="768" alt="image" src="https://user-images.githubusercontent.com/11302521/66441262-519eb580-ea36-11e9-9791-621e6d5d416d.png">